src/sentry/snuba/discover.py

```diff
@@ -0,0 +1,230 @@
+    raise ValueError("Unexpected condition format %s" % cond)
+
+
+def resolve_discover_aliases(snuba_args):
+    """
+    Resolve the public schema aliases to the discover dataset.
+
+    Returns a copy of the input structure, and includes a
+    `translated_columns` key containing the selected fields that need to
+    be renamed in the result set.
+    """
+    resolved = deepcopy(snuba_args)
+    translated_columns = {}
+    derived_columns = set()
+
+    selected_columns = resolved.get("selected_columns")
+    if selected_columns:
+        for (idx, col) in enumerate(selected_columns):
+            if isinstance(col, (list, tuple)):
+                # if list, means there are potentially nested functions and need to
+                # iterate and translate potential columns which is the same process
+                # as handling a condition.
+                selected_columns[idx] = resolve_condition(col, Dataset.Discover)
+                derived_columns.append(col[2])
+            else:
+                name = resolve_column(col)
+                selected_columns[idx] = name
+                translated_columns[name] = col
+        resolved["selected_columns"] = selected_columns
+    resolved["translated_columns"] = translated_columns
+
+    groupby = resolved.get("groupby")
+    if groupby:
+        for (idx, col) in enumerate(groupby):
+            name = col
+            if col not in derived_columns:
+                name = resolve_column(col)
+            groupby[idx] = name
+        resolved["groupby"] = groupby
+
+    aggregations = resolved.get("aggregations")
+    for aggregation in aggregations or []:
+        derived_columns.add(aggregation[2])
+        if isinstance(aggregation[1], six.string_types):
+            aggregation[1] = resolve_column(aggregation[1])
+        elif isinstance(aggregation[1], (set, tuple, list)):
+            aggregation[1] = [resolve_column(col) for col in aggregation[1]]
+    resolved["aggregations"] = aggregations
+
+    conditions = resolved.get("conditions")
+    if conditions:
+        for (i, condition) in enumerate(conditions):
+            replacement = resolve_condition(condition)
+            conditions[i] = replacement
+        resolved["conditions"] = list(filter(None, conditions))
+
+    # TODO add support for extracting having conditions.
+
+    orderby = resolved.get("orderby")
+    if orderby:
+        orderby = orderby if isinstance(orderby, (list, tuple)) else [orderby]
+        resolved_orderby = []
+
+        for field_with_order in orderby:
+            field = field_with_order.lstrip("-")
+            resolved_orderby.append(
+                u"{}{}".format(
+                    "-" if field_with_order.startswith("-") else "",
+                    field if field in derived_columns else resolve_column(field),
+                )
+            )
+        resolved["orderby"] = resolved_orderby
+    return resolved
+
+
+def transform_results(result, translated_columns):
```

Comment:
This seems almost identical to `transform_aliases_and_query` except it references the Discover names instead of the Event table names. Can we share some code?